packages/node_modules/@ciscospark/internal-plugin-metrics/src/metrics.js

```diff
@@ -99,6 +121,111 @@ const Metrics = SparkPlugin.extend({
       eventPayload: payload
     };
     return this.callDiagnosticEventsBatcher.request(event);
+  },
+
+  storeMetric(id, metricProps, metricType) {
+    const tuple = {metric: metricProps, type: metricType};
+    return this.metricMap.set(id, tuple);
+  },
+
+  getMetric(id) {
+    return this.metricMap.get(id);
+  },
+
+  deleteMetric(id) {
+    return this.metricMap.delete(id);
+  },
+
+  /**
+   * registers (stores) a metric if it does not exist
+   * @param {string} eventName
+   * @param {string} id
+   * @param {Object} props
+   * @param {string} type
+   * @returns {bool}
+   */
+  _registerMetric(eventName, id, props, type) {
+    if (!this.getMetric(id)) {
+      // hash name and props into table/object
+      const payload = this._createPayload(eventName, props);
+      return this.storeMetric(id, payload, type);
+    }
+    return false;
+  },
+
+  /**
+  * Registers a client metric to be used to later update or call finish method
+  * @param {string} eventName name of the metric
+  * @param {string} id unique identifier
+  * @param {Object} props to populate the metric with
+  * @returns {bool} true if metric registered, false otherwise
+  */
+  registerClientMetric(eventName, id, props) {
+    return this._registerMetric(eventName, id, props, this.CLIENT_METRIC);
+  },
+
+  /**
+  * Registers a metric to be used to later update or call finish method
+  * @param {string} eventName name of the metric
+  * @param {string} id unique identifier
+  * @param {Object} props to populate the metric with
+  * @returns {bool} true if metric registered, false otherwise
+  */
+  registerMetric(eventName, id, props) {
+    return this._registerMetric(eventName, id, props, this.SDK_METRIC);
+  },
+
+  /**
+   * Updates a registered metric, if it exists
+   * @param {string} id lookup key for metric
+   * @param {Object} metricProps to be used for updating metric
+   * @returns {bool} true if the metrics exists and was updated, false otherwise
+   */
+  updateRegisteredMetric(id, metricProps) {
+    const storedMetric = this.getMetric(id);
+    const props = storedMetric ? storedMetric.metric : null;
+    const type = storedMetric ? storedMetric.type : null;
+
+    if (!props || !type || !metricProps) {
+      return false;
+    }
+
+    // update metric
+    Object.assign(props.fields, metricProps.fields);
+
+    return this.storeMetric(id, props, type);
+  },
+
+  metricIsType(id, type) {
+    return this.getMetric(id) && this.getMetric(id).type === type;
```

Comment:
store the result of `getMetric()` so that you're only doing one lookup. you could also use lodash's `get()`

```javascript
return get(this.getMetric(id), `type`) === type;
```

Reply:
done
Not sure there is a big cost differential between to two O(1) lookups versus comparing local variable, and one uses more bytes, but is a small enough change.